src/main/java/it/gov/pagopa/admissibility/service/onboarding/AuthoritiesDataRetrieverServiceImpl.java

```diff
@@ -47,45 +70,70 @@ public Mono<OnboardingDTO> retrieve(OnboardingDTO onboardingRequest, InitiativeC
+                    .zipWith(userFiscalCodeService.getUserFiscalCode(onboardingRequest.getUserId()))
+                    .flatMap(t -> invokePdndServices(t.getT1(), onboardingRequest, t.getT2(), pdndServicesInvocation, message));
         }
+
         return Mono.just(onboardingRequest);
     }
 
-    private static Random userIdBasedIntegerGenerator(OnboardingDTO onboardingRequest) {
-        @SuppressWarnings("squid:S2245")
-        Random random = new Random(onboardingRequest.getUserId().hashCode());
-        return random;
+    private Mono<OnboardingDTO> invokePdndServices(String accessToken, OnboardingDTO onboardingRequest, String fiscalCode, PdndServicesInvocation pdndServicesInvocation, Message<String> message) {
+        Mono<Optional<Object>> inpsInvocation = pdndServicesInvocation.requireInpsInvocation()
+                ? Mono.just(Optional.of("TODO Call INPS Service"))
+                : Mono.just(Optional.empty());
+
+        Mono<Optional<RispostaE002OKDTO>> anprInvocation = pdndServicesInvocation.requireAnprInvocation()
+                ? residenceAssessmentService.getResidenceAssessment(accessToken, fiscalCode).map(Optional::of)
+                .onErrorResume(AnprDailyRequestLimitException.class, e -> {
+                    log.debug("[ONBOARDING_REQUEST][RESIDENCE_ASSESSMENT] ", e);
+                    return Mono.just(Optional.empty());
+                })
+                : Mono.just(Optional.empty());
+
+        return inpsInvocation
+                .zipWith(anprInvocation)
+                // Handle reschedule of failed invocation, storing the successful if any
+                .mapNotNull(t -> {
+                    extractPdndResponses(
+                            onboardingRequest,
+                            pdndServicesInvocation,
+                            t.getT1().orElse(null),
+                            t.getT2().orElse(null));
+
+                    if ((!pdndServicesInvocation.requireInpsInvocation() || t.getT1().isPresent())
+                            &&
+                            (!pdndServicesInvocation.requireAnprInvocation() || t.getT2().isPresent())) {
+                        return onboardingRequest;
+                    } else {
+                        rischeduleOnboardingRequest(onboardingRequest, message);
+                        return null;
+                    }
+                });
+    }
+
+    private void extractPdndResponses(OnboardingDTO onboardingRequest, PdndServicesInvocation pdndServicesInvocation, Object inpsResponse, RispostaE002OKDTO anprResponse) {
+        if (inpsResponse != null && pdndServicesInvocation.getIsee) {
+            onboardingRequest.setIsee(BigDecimal.TEN);  // TODO
+        }
+        if (anprResponse != null) {
+            TipoResidenzaDTO residenza = anprResponse.getListaSoggetti().getDatiSoggetto().get(0).getResidenza().get(0);
+            TipoGeneralitaDTO generalita = anprResponse.getListaSoggetti().getDatiSoggetto().get(0).getGeneralita();
```

Comment:
use english name for variables and check if not null (moving the get operation inside the subsequent if)

Reply:
Add a TODO to remember to ask what to do if the info is not present